java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java

```diff
@@ -0,0 +1,201 @@
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+import static org.springframework.http.HttpStatus.OK;
+
+import com.amido.stacks.core.api.annotations.CreateAPIResponses;
+import com.amido.stacks.core.api.annotations.DeleteAPIResponses;
+import com.amido.stacks.core.api.annotations.SearchAPIResponses;
+import com.amido.stacks.core.api.annotations.UpdateAPIResponses;
+import com.amido.stacks.core.api.dto.response.ResourceCreatedResponse;
+import com.amido.stacks.core.api.dto.response.ResourceUpdatedResponse;
+import com.amido.stacks.workloads.menu.api.v1.dto.request.CreateMenuRequest;
+import com.amido.stacks.workloads.menu.api.v1.dto.request.UpdateMenuRequest;
+import com.amido.stacks.workloads.menu.api.v1.dto.response.MenuDTO;
+import com.amido.stacks.workloads.menu.api.v1.dto.response.SearchMenuResult;
+import com.amido.stacks.workloads.menu.commands.DeleteMenuCommand;
+import com.amido.stacks.workloads.menu.commands.MenuCommand;
+import com.amido.stacks.workloads.menu.commands.UpdateMenuCommand;
+import com.amido.stacks.workloads.menu.domain.Menu;
+import com.amido.stacks.workloads.menu.exception.MenuNotFoundException;
+import com.amido.stacks.workloads.menu.handlers.CreateMenuHandler;
+import com.amido.stacks.workloads.menu.handlers.DeleteMenuHandler;
+import com.amido.stacks.workloads.menu.handlers.UpdateMenuHandler;
+import com.amido.stacks.workloads.menu.mappers.MenuMapper;
+import com.amido.stacks.workloads.menu.mappers.RequestToCommandMapper;
+import com.amido.stacks.workloads.menu.mappers.SearchMenuResultItemMapper;
+import com.amido.stacks.workloads.menu.service.MenuQueryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestAttribute;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping(path = "/v1/menu", produces = MediaType.APPLICATION_JSON_VALUE + " ; charset=utf-8")
+@RestController
+public class MenuController {
+
+  private CreateMenuHandler createMenuHandler;
```

Comment:
I suspect IntelliJ is telling you all of these can be `final`?

Reply:
yes it does in big Yellow ...

Reply:
fixed